shop/src/main/java/ro/msg/learning/shop/repository/StockRepository.java

```diff
@@ -0,0 +1,44 @@
+package ro.msg.learning.shop.repository;
+
+import jakarta.persistence.criteria.Predicate;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.domain.StockId;
+import ro.msg.learning.shop.dto.ProductIdAndQuantity;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+@Repository
+public interface StockRepository extends JpaRepository<Stock, StockId>, JpaSpecificationExecutor<Stock> {
+    default List<UUID> findLocationIdsByProductIdsAndStock(List<ProductIdAndQuantity> productIdAndQuantityList, List<Product> products) {
+        return this.findAll((Specification<Stock>) (root, query, criteriaBuilder) -> {
+
+            Predicate[] predicates = new Predicate[productIdAndQuantityList.size()];
+
+            for (int i = 0; i < productIdAndQuantityList.size(); i++) {
```

Comment:
can this method be a little bit refactored?

Reply:
i dont know how to use streaming here because i have 2 different list and i need to match elements from the same index

Reply:
refactor does not mean make shorter by using streams. It means to group logical components and extract them in separate methods